### ISSUE TITLE
Conda 4.6.7 regression with Pip dependencies in environment.yml

### ISSUE THREAD

**Issue by bollard** [conda/conda#8456]
## Current Behavior

From Conda 4.6.7 onwards, packages required under the `pip` dependencies in an `environment.yml` no longer appear to have access to packages previously installed in the main `dependencies` section. This works in 4.6.4 (4.6.5 and 4.6.6 aren't available in `defaults` so I'm unable to check them).

I have tried enabling `pip_interop_enabled` but this did not help.

I'm on Windows 10.

### Steps to Reproduce

Create the below in `environment.yml`

```
name: foo
dependencies:
  - python=3.6
  - cython
  - pip:
    - pyjnius
```

NB: PyJnius required Cython for setup.

Now run `conda env create --file=environment.yml` and observe the following error

```
(base) D:\Research\python\trunk>conda env create --file=environment.yml
Collecting package metadata: done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.6.7
  latest version: 4.6.8

Please update conda by running

    $ conda update -n base -c defaults conda


Preparing transaction: done
Verifying transaction: done
Executing transaction: done

CondaEnvException: Pip subcommand failed with
output: Collecting pyjnius (from -r D:\Research\python\trunk\condaenv.8a0zzlcu.requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/b6/57/c90acf31322e6417f06c90410dbfcb149633a6006b7efbf99dfebe177c1f/pyjnius-1.2.0.tar.gz
    Complete output from command python setup.py egg_info:


    You need Cython to compile Pyjnius.


    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\FIRST~1.LAST\AppData\Local\Temp\pip-install-bxl5ez_l\pyjnius\setup.py", line 65, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'

    ----------------------------------------

error: The system cannot find the path specified.
Command "python setup.py egg_info" failed with error code 1 in C:\Users\FIRST~1.LAST\AppData\Local\Temp\pip-install-bxl5ez_l\pyjnius\
```

## Expected Behavior

Under Conda 4.6.4, Cython was available to Pyjnius

### Steps to Reproduce

```
conda env remove --name=foo
conda install conda=4.6.4
conda env create --file=environment.yml
```

Produces, as expected...

```
(base) D:\Research\python\trunk>echo "PATH: D:\Miniconda3;D:\Miniconda3\Library\mingw-w64\bin;D:\Miniconda3\Library\usr\bin;D:\Miniconda3\Library\bin;D:\Miniconda3\Scripts;D:\Miniconda3\bin;D:\Miniconda3\condabin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\ProgramData\chocolatey\bin;C:\Program Files\MATLAB\R2017b\bin;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Java\jdk1.8.0_201\jre\bin\server;C:\Program Files\Java\jdk1.8.0_201\bin;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files\TortoiseSVN\bin;C:\Users\first.last\AppData\Local\Microsoft\WindowsApps;
"PATH: D:\Miniconda3;D:\Miniconda3\Library\mingw-w64\bin;D:\Miniconda3\Library\usr\bin;D:\Miniconda3\Library\bin;D:\Miniconda3\Scripts;D:\Miniconda3\bin;D:\Miniconda3\condabin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\ProgramData\chocolatey\bin;C:\Program Files\MATLAB\R2017b\bin;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Java\jdk1.8.0_201\jre\bin\server;C:\Program Files\Java\jdk1.8.0_201\bin;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Program Files\TortoiseSVN\bin;C:\Users\first.last\AppData\Local\Microsoft\WindowsApps;

(base) D:\Research\python\trunk>D:\Miniconda3\envs\foo\python.exe D:\Miniconda3\envs\foo\Scripts\pip-script.py --disable-pip-version-check install -r D:\Research\python\trunk\condaenv.46glxo65.requirements.txt
Collecting pyjnius (from -r D:\Research\python\trunk\condaenv.46glxo65.requirements.txt (line 1))
Collecting six>=1.7.0 (from pyjnius->-r D:\Research\python\trunk\condaenv.46glxo65.requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Requirement already satisfied: cython in d:\miniconda3\envs\foo\lib\site-packages (from pyjnius->-r D:\Research\python\trunk\condaenv.46glxo65.requirements.txt (line 1)) (0.29.6)
Installing collected packages: six, pyjnius
Successfully installed pyjnius-1.2.0 six-1.12.0
#
# To activate this environment, use
#
#     $ conda activate foo
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
(base) D:\Research\python\trunk>conda info

     active environment : base
    active env location : D:\Miniconda3
            shell level : 1
       user config file : C:\Users\first.last\.condarc
 populated config files : C:\Users\first.last\.condarc
          conda version : 4.6.7
    conda-build version : not installed
         python version : 3.7.0.final.0
       base environment : D:\Miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : D:\Miniconda3\pkgs
                          C:\Users\first.last\.conda\pkgs
                          C:\Users\first.last\AppData\Local\conda\conda\pkgs
       envs directories : D:\Miniconda3\envs
                          C:\Users\first.last\.conda\envs
                          C:\Users\first.last\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.6.7 requests/2.21.0 CPython/3.7.0 Windows/10 Windows/10.0.17763
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
D:\Research\python\trunk>conda config --show-sources
==> C:\Users\first.last\.condarc <==
pip_interop_enabled: False
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
(base) D:\Research\python\trunk>conda list --show-channel-urls
# packages in environment at D:\Miniconda3:
#
# Name                    Version                   Build  Channel
arctic                    1.74.0                   pypi_0    pypi
asn1crypto                0.24.0                   py37_0    defaults
backcall                  0.1.0                    py37_0    defaults
ca-certificates           2019.1.23                     0    defaults
certifi                   2019.3.9                 py37_0    defaults
cffi                      1.11.5           py37h74b6da3_1    defaults
chardet                   3.0.4                    py37_1    defaults
colorama                  0.4.1                    py37_0    defaults
conda                     4.6.7                    py37_0    defaults
conda-env                 2.6.0                         1    defaults
console_shortcut          0.1.1                         3    defaults
cryptography              2.3.1            py37h74b6da3_0    defaults
decorator                 4.4.0                    py37_1    defaults
enum-compat               0.0.2                    pypi_0    pypi
h5py                      2.9.0                    pypi_0    pypi
hdf5storage               0.1.15                   pypi_0    pypi
idna                      2.8                      py37_0    defaults
ipython                   7.4.0            py37h39e3cac_0    defaults
ipython_genutils          0.2.0                    py37_0    defaults
jedi                      0.13.3                   py37_0    defaults
jupyter_client            5.2.4                    py37_0    defaults
jupyter_core              4.4.0                    py37_0    defaults
libsodium                 1.0.16               h9d3ae62_0    defaults
lz4                       2.1.6                    pypi_0    pypi
menuinst                  1.4.14           py37hfa6e2cd_0    defaults
mockextras                1.0.2                    pypi_0    pypi
msvc_runtime              1.0.1                    vc14_0    defaults
numpy                     1.16.2                   pypi_0    pypi
openssl                   1.0.2p               hfa6e2cd_0    defaults
pandas                    0.24.2                   pypi_0    pypi
parso                     0.3.4                    py37_0    defaults
pickleshare               0.7.5                    py37_0    defaults
pip                       19.0.3                   py37_0    defaults
prompt_toolkit            2.0.9                    py37_0    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.19                     py37_0    defaults
pygments                  2.3.1                    py37_0    defaults
pymongo                   3.7.2                    pypi_0    pypi
pyopenssl                 19.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.0                hea74fb7_0    defaults
python-dateutil           2.8.0                    py37_0    defaults
pytz                      2018.9                   pypi_0    pypi
pywin32                   223              py37hfa6e2cd_1    defaults
pyzmq                     17.1.2           py37hfa6e2cd_0    defaults
requests                  2.21.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
setuptools                40.8.0                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
teamcity-messages         1.24                     pypi_0    pypi
tornado                   5.1.1            py37hfa6e2cd_0    defaults
traitlets                 4.3.2                    py37_0    defaults
tzlocal                   1.5.1                    pypi_0    pypi
urllib3                   1.24.1                   py37_0    defaults
vc                        14                   h0510ff6_3    defaults
vs2015_runtime            14.15.26706          h3a45250_0    defaults
wcwidth                   0.1.7                    py37_0    defaults
wheel                     0.33.1                   py37_0    defaults
win_inet_pton             1.1.0                    py37_0    defaults
wincertstore              0.2                      py37_0    defaults
yaml                      0.1.7                hc54c509_2    defaults
zeromq                    4.2.5                he025d50_1    defaults
```
</p></details>

**Comment by tjd2002** [comment]
Possibly related to #8404, which was fixed in 4.6.9?

Can you replicate on a more recent conda?

**Comment by bollard** [comment]
Yes, can confirm issue now fixed on 4.6.16. Thanks
